src/elements/ExactCFbend.H

```diff
@@ -0,0 +1,665 @@
+            particle(5) = tout;
+            particle(6) = ptout;
+
+            zeval += 0_prt;
+        }
+
+        /** This pushes a particle in an ExactMultipole element through
+         *  the symplectic map associated with H_2 in the Hamiltonian
+         *  splitting H = H_1 + H_2.  This is the exact kick obtained
+         *  in the ideal multipole field.
+         *
+         * @param tau Map step size in m
+         * @param[in,out] particle particle phase space 6-vector
+         * @param[in,out] zeval Longitudinal on-axis location in m
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void map2 (amrex::ParticleReal const tau,
+                   amrex::SmallVector<amrex::ParticleReal, 6, 1> & particle,
+                   amrex::ParticleReal & zeval) const
+        {
+            using namespace amrex::literals; // for _rt and _prt
+            using namespace amrex::Math;  // for powi
+
+            amrex::ParticleReal const x = particle(1);
+            amrex::ParticleReal const px = particle(2);
+            amrex::ParticleReal const y = particle(3);
+            amrex::ParticleReal const py = particle(4);
+            amrex::ParticleReal const t = particle(5);
+            amrex::ParticleReal const pt = particle(6);
+
+            amrex::ParticleReal xout = x;
+            amrex::ParticleReal pxout = px;
+            amrex::ParticleReal yout = y;
+            amrex::ParticleReal pyout = py;
+            amrex::ParticleReal tout = t;
+            amrex::ParticleReal ptout = pt;
+
+#if AMREX_DEVICE_COMPILE
+            amrex::ParticleReal* k_normal = m_k_normal_d_data;
+            amrex::ParticleReal* k_skew = m_k_skew_d_data;
+#else
+            amrex::ParticleReal* k_normal = m_k_normal_h_data;
+            amrex::ParticleReal* k_skew = m_k_skew_h_data;
+#endif
+
+            // defined dimensionless variables scaled by radius of curvature
+            amrex::ParticleReal rho = 1_prt + x / m_rc;
+            amrex::ParticleReal y_scl = y / m_rc;
+            amrex::ParticleReal ln_rho = log(rho);
+
+            // initialize the momentum kick
+            amrex::ParticleReal dpx = 0_prt;
+            amrex::ParticleReal dpy = 0_prt;
+
+            // loop over array of multipole coefficients
+            for (int m = 1; m < m_ncoef; m++) {
+
+               // normalize units to MAD-X convention if needed
+               amrex::ParticleReal kn = m_unit == 1 ? k_normal[m] / m_brho : k_normal[m];
+               amrex::ParticleReal ks = m_unit == 1 ? k_skew[m] / m_brho : k_skew[m];
+
+               int n = m+1;  // multipole order
+
+               //amrex::Print() << "n, kn, ks before scaling: " << n << ", " << kn << ", " << ks << "\n";
+
+               // define dimensionless multipole coefficients (scaled by radius of curvature)
+               kn = std::pow(m_rc, m) * kn; // unable to use powi here
+               ks = std::pow(m_rc, m) * ks; // unable to use powi here
```

Comment:
Yes, only works for (compile-time) constant `m` :)
```suggestion
               kn = std::pow(m_rc, m) * kn;
               ks = std::pow(m_rc, m) * ks;
```